SyncConnectionStateErrorReset/SyncConnectionStateErrorReset/AppViews.swift

```diff
@@ -0,0 +1,265 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import SwiftUI
+import RealmSwift
+
+struct InitialView: View {
+    @ObservedObject var realmManager = RealmManager()
+    @ObservedObject var app: RealmSwift.App
+
+    var body: some View {
+        // This observes changes on the App, which includes User State changes,
+        // and will trigger if the user state changes during the app cycle of the App.
+        switch app.currentUser?.state {
+        case .loggedIn:
+            if let user = app.currentUser {
+                if let realm = realmManager.realm {
+                    KioskView(user: user)
+                        .environment(\.realmConfiguration, realm.configuration)
+                        .toolbar {
+                            ToolbarItemGroup(placement: .bottomBar) {
+                                HStack {
+                                    VStack {
+                                        Text("Session State")
+                                        Text("\(realmManager.sessionState?.literal ?? "")")
+                                    }
+                                    Spacer()
+                                    VStack {
+                                        Text("Connection State")
+                                        Text("\(realmManager.connectionState?.literal ?? "")")
+                                    }
+                                }
+                            }
+                        }
+                } else {
+                    ProgressView()
+                        .onAppear {
+                            Task {
+                                do {
+                                    // Opens the realm asynchronously, and waits for the data to be downloaded.
+                                    try await realmManager.openRealmWithUser(user)
+                                } catch {
+                                    print("Error while opening realm \(error)")
+                                }
+                            }
+                        }
+                }
+
+            }
+        case .loggedOut, .removed, .none:
+            LoginView()
+                .padding()
+        default: EmptyView() // Unreachable
+        }
+    }
+}
+
+// Login/Sign Up View.
+struct LoginView: View {
+    @ObservedObject var appAuthManager = AppAuthManager()
+
+    @State var username: String = "john@doe.com"
+    @State var password: String = "123456"
+
+    var body: some View {
+        ZStack {
+            VStack {
+                Spacer()
+                VStack {
+                    TextField("Username/Email", text: $username)
+                        .disableAutocorrection(true)
+                        .textInputAutocapitalization(.none)
```

Comment:
This and the following `SecureField` are autocapitalizing in spite of calling this instance method. I checked into it and it looks like there are two options:

- `.autocapitalization(.none)` (some of the info I found suggested this is the older method)
- `.textInputAutocapitalization(.never)` (some of the info I found suggested this replaced .autocapitalization in iOS 15)

According to [the API docs](https://developer.apple.com/documentation/swiftui/textinputautocapitalization), `.textInputAutocapitalization` doesn't seem to have a `.none` option - if we want to use this instance method, I think we would specify `.never`, instead. It looks like `.none` is only a meaningful option for the older `.autocapitalization()` method. The compiler says `.none` here in `.textInputAutocapitalization` is the absence of a value, so I guess that's why it's not complaining, but it isn't actually changing the autocapitalization behavior.

We should probably change this to `.textInputAutocapitalization(.never)` (and the same in line 89 below).